gammapy/utils/fitting/tests/test_scipy.py

```diff
@@ -4,17 +4,19 @@
 from .. import Parameter, Parameters, optimize_scipy, confidence_scipy
 
 
-def fcn(parameters):
-    x = parameters["x"].value
-    y = parameters["y"].value
-    z = parameters["z"].value
-    x_opt, y_opt, z_opt = 2, 3e5, 4e-5
-    x_err, y_err, z_err = 0.2, 3e4, 4e-6
-    return (
-        ((x - x_opt) / x_err) ** 2
-        + ((y - y_opt) / y_err) ** 2
-        + ((z - z_opt) / z_err) ** 2
-    )
+class MyDataset:
```

Comment:
Again?

I think the fitting backend unit tests should be re-structured, especially for iminuit, no?

Open a reminder issue for v0.13?